firewalldb/kvstores.go

```diff
@@ -9,8 +9,11 @@ import (
 
 /*
 The KVStores are stored in the following structure in the KV db. Note that
-the `perm` and `temp` buckets are identical in structure. The only difference is
-that the `temp` bucket is cleared on restart of the db.
+the `perm` and `temp` buckets are identical in structure. The only difference
+is that the `temp` bucket is cleared on restart of the db. The reason persisting
+the temporary store changes instead of just keeping an in-memory store is that
+we can then guarantee idempotency if changes are made to both the permanent and
```

Comment:
nit: not 100% sure but wouldn't it be called atomicity instead of idempotency in this context?

Reply:
indeed! you are correct - will update